lib/fs.js

```diff
@@ -1097,7 +1097,9 @@ function writeAll(fd, buffer, offset, length, position, callback) {
       } else {
         offset += written;
         length -= written;
-        position += written;
+        if (position !== null) {
```

Comment:
I think the check needs to be `if (typeof position === 'number') {`, else `undefined` or `false` will still trigger the bad behavior.  It's also the condition that the documentation for `fs.write()` mentions.